### PR TITLE
Use curl

### PR DESCRIPTION
**Pull request by shanepelletier**
Note that curl supports a "-v" command-line switch which displays the same information we used to custom-print (albeit in a different way), so handling "-v" separately isn't necessary.

### CODE REVIEW

File: surl/__init__.py
Comment (maxiberta) [inline surl/__init__.py:457]: Why redirect stderr to stdout? And, should we handle errors i.e. pass `check=True` instead of failing silently?

Comment (shanepelletier) [reply surl/__init__.py:457]: Redirecting stderr to stdout means that we don't fail silently; any failures (e.g., unable to connect) are reported by curl using stderr. We could add error handling ourselves but that means we have to decide what error messages to use rather than letting curl do the thinking for us.

Comment (shanepelletier) [reply surl/__init__.py:457]: Haha, same question Maxi asked. Without that redirect, stderr just gets sent to the void, never to be seen again. With it, the output is at least forwarded to surl. There might be a way to get stderr from a subprocess if we really need it. I'll add the return code change.

File: surl/__init__.py
Comment (Roadmaster) [inline surl/__init__.py:459]: Hi, I'm fairly certain this _noop method and the try/except below to flush stdout are not needed if we're letting curl handle all output; this seems to have been needed because the previous implementation did streaming read of the response and corresponding write by chunks to sys.stdout (see above diff lines 536-538 which were removed).

If you want to confirm, try downloading a snap, the issue that the above code fixed was https://github.com/canonical/surl/issues/8.

**Review by maxiberta — APPROVED** [review]
+1!

File: surl/__init__.py
Comment (Roadmaster) [reply surl/__init__.py]: I see - I think we want this then:
```
include sys
subprocess.run(arguments, stderr=sys.stderr)
```
My thinking here is that it'll be confusing or breaky if someone expects surl to just respect curl's handling of stdout and stderr, and then we pipe stderr output to stdout. So by doing the above, I *think* we're just saying "ok, don't eat stderr output, send it to the actual stderr file descriptor.

I think this is not a blocker, if you have time to check whether my speculation is correct and feel like changing this that'd be OK but if you prefer to leave as is, that works for me. And this fix can be done in a subsequent PR so we don't hold this awesome change on a triviality.

**Comment by Roadmaster** [comment]
+1 from me, there are two outstanding suggested changes, I'll leave it to you to either:

1. implement before merging this
2. merge this as-is and then later do the changes in another pr
3. totally ignore me ;)

**Review by Roadmaster — APPROVED** [review]
+1

**Pull request merged**
